core/citrus-base/src/main/java/org/citrusframework/functions/core/AdvancedRandomNumberFunction.java

```diff
@@ -134,7 +134,7 @@ private String getRandomNumber(int decimalPlaces,
 
         return decimalPlaces == 0 ?
             format("%s", randomValue.longValue()) :
-            format(format("%%.%sf", decimalPlaces), randomValue.doubleValue());
+            format("%." + decimalPlaces + "f", randomValue.doubleValue());
```

Comment:
I think this is not the same functionality. or is it? I don't think so, because now you're printing the `decimalPlaces` literally, instead of repeated times.

Reply:
It is the same and it is meant as such. Decimal places should be printed literally to form the correct format.

```java
System.out.println(format(format("%%.%sf", 4), 1.2345678));
System.out.println(format("%." + 4 + "f", 1.2345678));
```
1.2346
1.2346
```

```

Reply:
thanks for clarification!